rigex-front-end/src/components/auth/signUpForm.js

```diff
@@ -0,0 +1,203 @@
+import React, { useState, useEffect } from 'react';
+import { gql, useMutation } from '@apollo/client';
+import { useDispatch } from 'react-redux';
+
+import Button from '@material-ui/core/Button';
+import IconButton from '@material-ui/core/IconButton';
+import Input from '@material-ui/core/Input';
+import FormControl from '@material-ui/core/FormControl';
+import InputLabel from '@material-ui/core/InputLabel';
+import Mail from '@material-ui/icons/Email';
+import Lock from '@material-ui/icons/Lock';
+import VisibilityOutlined from '@material-ui/icons/VisibilityOutlined';
+import VisibilityOffOutlined from '@material-ui/icons/VisibilityOffOutlined';
+import Grid from '@material-ui/core/Grid';
+
+import './signUpForm.scss';
+import { signUpActions } from '../../store/signUp/action';
+import logo from '../../assets/large-logo.png';
+import history from '../../utils/history';
+import routePaths from '../../constants/routePaths';
+
+const SIGN_UP_USER = gql`
+  mutation CreateAccount($model: UserInput!) {
+    createAccount(model: $model) {
+      id
+    }
+  }
+`;
+
+const SignUpForm = () => {
+  const [userState, setUserState] = useState({
+    email: '',
+    password: '',
+    showPassword: false,
+  });
+  const [errorMessage, setErrorMessage] = useState('');
+  const [helpMessage, setHelpMessage] = useState('');
+  const [CreateAccount, { loading, error, data }] = useMutation(SIGN_UP_USER);
+  const dispatch = useDispatch();
+  // eslint-disable-next-line no-useless-escape
+  function checkPwd(str) {
```

Comment:
I think this function could have a more descriptive name and be placed in a util folder.